src/error.rs

```diff
@@ -39,7 +39,7 @@ pub enum Error {
     NoRecipients,
     NoUtxosSelected,
     OutputBelowDustLimit(usize),
-    InsufficientFunds,
+    InsufficientFunds(u64, u64), // amount needed, amount available
```

Comment:
Would it be better to make a struct for this pair?

Also, would it be better to call the second one "amount selected" instead of "amount available"? Are there cases where the wallet may have enough but coin selection doesn't choose the right combination of utxos?

Reply:
More than a struct you could just convert this to a named enum variant, like `InsufficientFunds { needed: u64, available: u64 }`

Reply:
Thanks! Updated accordingly.